### PR TITLE
[tests-only][full-ci] refactor profind response

### PR DESCRIPTION
**Pull request by nabim777**
## Description

In this PR,
- function `findEntryFromPropfindResponse` is refactor for strictly path checking
- method REPORT is removed from function `findEntryFromPropfindResponse` since there is separate function make for REPORT which is `findEntryFromSearchResponse`.

## Related Issue

Closes https://github.com/owncloud/ocis/issues/7006

## How Has This Been Tested?
- test environment:
- locally
- CI

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/SearchContext.php [hunk 1]
```diff
@@ -122,8 +122,7 @@ public function fileOrFolderInTheSearchResultShouldContainProperties(
 		$properties = $properties->getHash();
 		$fileResult = $this->featureContext->findEntryFromPropfindResponse(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SearchContext.php:123]: should it be?
```suggestion
		$fileResult = $this->featureContext->findEntryFromSearchResponse(
```

Comment (nabim777) [reply tests/acceptance/features/bootstrap/SearchContext.php]: I think yes

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -4881,19 +4881,23 @@ public function propfindResultShouldContainEntries(
 				continue;
 			}
 			if ($method === "REPORT") {
-				$fileFound = $this->findEntryFromSearchResponse(
+				$xmlItem = $this->findEntryFromSearchResponse(
 					$resource
 				);
+				if (!empty($xmlItem)) {
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/WebDav.php:4887]: will this condition work for the xml i think it was empty when we checked?

File: tests/acceptance/features/coreApiWebdavOperations/search.feature [hunk 3]
```diff
@@ -186,23 +186,23 @@ Feature: Search
...
+      | name                  | value                                                                                             |
+      | oc:fileid             | \d*                                                                                               |
+      | oc:permissions        | ^(RDNVW\|RMDNVW)$                                                                                 |
+      | d:getlastmodified     | ^[MTWFS][uedhfriatno]{2},\s(\d){2}\s[JFMAJSOND][anebrpyulgctov]{2}\s\d{4}\s\d{2}:\d{2}:\d{2} GMT$ |
+      | d:getetag             | ^\"[a-f0-9:\.]{1,32}\"$                                                                           |
+      | d:getcontenttype      | text\/plain                                                                                       |
+      | oc:size               | 15                                                                                                |
+      | oc:owner-id           | %username%                                                                                        |
+      | oc:owner-display-name | %displayname%                                                                                     |
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiWebdavOperations/search.feature:205]: also please verify that this is the expected behaviour of ocis?

Comment (nabim777) [reply tests/acceptance/features/coreApiWebdavOperations/search.feature:205]: ok

File: tests/acceptance/features/bootstrap/SearchContext.php [hunk 4]
```diff
@@ -133,36 +133,28 @@ public function fileOrFolderInTheSearchResultShouldContainProperties(
...
-					break;
-				}
+			$fileResultProperty = $fileResult->xpath("d:propstat//" . $property['name']);
+			if ($fileResultProperty) {
+				Assert::assertMatchesRegularExpression(
+					"/" . $property['value'] . "/",
+					\trim((string)$fileResultProperty[0])
+				);
+			} else {
+				throw new Error("could not find property '" . $property['name'] . "'");
```
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/SearchContext.php:156]:
```suggestion
				throw new Error("Could not find property '" . $property['name'] . "'");
```

**Comment by amrita-shrestha** [comment]
did you try testing step for false positive and false negative results?

**Comment by nabim777** [comment]
> did you try testing step for false positive and false negative results?

yes I try

File: tests/acceptance/features/coreApiWebdavOperations/search.feature (hunk 3 above)
Comment (nabim777) [reply tests/acceptance/features/coreApiWebdavOperations/search.feature:205]: can we do these changes in another PR?

File: tests/acceptance/features/bootstrap/SearchContext.php [hunk 5]
```diff
@@ -133,36 +133,28 @@ public function fileOrFolderInTheSearchResultShouldContainProperties(
...
-				}
+			$fileResultProperty = $fileResult->xpath("d:propstat//" . $property['name']);
+			if ($fileResultProperty) {
+				Assert::assertMatchesRegularExpression(
+					"/" . $property['value'] . "/",
+					\trim((string)$fileResultProperty[0])
+				);
+			} else {
+				throw new Error("Could not find property '" . $property['name'] . "'");
 			}
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SearchContext.php:157]:
```suggestion
				return Assert::assertMatchesRegularExpression(
					"/" . $property['value'] . "/",
					\trim((string)$fileResultProperty[0])
				);
			}
			throw new Error("Could not find property '" . $property['name'] . "'");
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 6]
```diff
@@ -4881,19 +4881,18 @@ public function propfindResultShouldContainEntries(
 				continue;
 			}
 			if ($method === "REPORT") {
-				$fileFound = $this->findEntryFromSearchResponse(
+				$xmlItem = $this->findEntryFromSearchResponse(
 					$resource
 				);
+				$fileFound = !\is_bool($xmlItem) ? $xmlItem->xpath("d:propstat//oc:name") : $xmlItem;
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDav.php:4887]:
```suggestion
				$fileFound = $this->findEntryFromSearchResponse(
					$resource
				);
```

Comment (SagarGi) [reply tests/acceptance/features/bootstrap/WebDav.php:4887]: I think the function return mixed type where string|bool|object. May be we check all of those as per required.

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/WebDav.php:4887]: so does this function `$fileFound = $this->findEntryFromPropfindResponse(` just below. Should be fine I think

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 7]
```diff
@@ -5387,7 +5371,7 @@ public function findEntryFromSearchResponse(
 			}
 			$resourcePath = \rawurldecode($resourcePath);
 			if ($entryNameToSearch === $resourcePath) {
-				return $resourcePath;
+				return $item;
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDav.php:5374]: why return item (SimpleXmlElement) we want to return `$resourcePath`, don't we?

File: tests/acceptance/features/bootstrap/SearchContext.php (hunk 5 above)
Comment (nabim777) [reply tests/acceptance/features/bootstrap/SearchContext.php:157]: It is a step definition i think it is not to do a return

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/SearchContext.php:157]: returning should not matter. It's just to break the code

**Review by saw-jan — APPROVED** [review]
LGTM

**Comment by saw-jan** [comment]
@nabin, some tests got passed
https://drone.owncloud.com/owncloud/ocis/27307/72/5

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

**Review by saw-jan — APPROVED** [review]

**Pull request merged**